test/common/event/file_event_impl_test.cc

```diff
@@ -439,6 +439,41 @@ TEST_F(FileEventImplTest, RegisterIfEmulatedEdge) {
   dispatcher_->run(Event::Dispatcher::RunType::NonBlock);
 }
 
+#ifndef WIN32 // This is a test case that clarifies the event behavior.
+// Read and Write events are activated by the kernel before invoke connect() or listen().
+TEST_P(FileEventImplActivateTest, EventTriggeredBeforeConnect) {
+  os_fd_t fd = os_sys_calls_.socket(domain(), SOCK_STREAM, 0).return_value_;
```

Comment:
So this test is about creation of stream sockets instead of the socketpairs used elsewhere in this test file?

Can you add some comments as to why covering this case is important and how we should expect it to differ from the socketpair case?

Reply:
> So this test is about creation of stream sockets instead of the socketpairs used elsewhere in this test file?

Right. This is the stream socket that OS provides. 
But the rest of the file is not socket pair. All this files are OS sockets.

Reply:
> Can you add some comments as to why covering this case is important and how we should expect it to differ from the socketpair case?

As we discussed, we can either cover this sequence in connectionimpl, or we should complete avoid this sequence in prod and test code.
This goal of the test case is to explain the necessity of the next step.